security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -648,9 +649,13 @@ func (sc *SecretManagerClient) generateNewSecret(resourceName string) (*security
 	}, nil
 }
 
-var rotateTime = func(secret security.SecretItem, graceRatio float64) time.Duration {
+var rotateTime = func(secret security.SecretItem, graceRatio float64, graceRatioJitter float64) time.Duration {
 	secretLifeTime := secret.ExpireTime.Sub(secret.CreatedTime)
-	gracePeriod := time.Duration((graceRatio) * float64(secretLifeTime))
+	jitterGraceRatio := graceRatio + (rand.Float64() * graceRatioJitter)
```

Comment:
This can only add time, can we make it +/- some %?

Reply:
Done!